src/canvas/Renderer.tsx

```diff
@@ -62,13 +63,35 @@ export class Renderer implements IRenderer {
 
```

Comment:
While testing, i found that weird things tended to happen when starting a path close to an existing path. I thought maybe we should to some checks on mousedown event, and possibly snap the first point of currPath to an endpoint or to a point on path right then and there to avoid problems later.

Reply:
I updated when we snap the start of a path to be on mouse down instead of endTracing. I don't think its very noticeable to the user, so I think it is acceptable to do while they are tracing.